README.md

```diff
@@ -1,89 +1,162 @@
+* The project is developed using [ComputeCpp](https://developer.codeplay.com)
+ using Ubuntu 16.04 on Intel CPU and Intel GPU.
+* A BLAS library, such as OpenBLAS, is also required by the project to verify the test results
+
+
+## Supported Platforms
+
+The master branch of SYCL-BLAS is regularly tested with the "Supported" hardware 
+listed on [the ComputeCpp Supported Platforms page](https://developer.codeplay.com/products/computecpp/ce/guides/platform-support).
+
+SYCL-BLAS may also work on other hardware and platforms assuming they implement 
+ SPIR or SPIR-V support.
+
+## Getting Started with SYCL-BLAS
+
+### Pre-requisites
+
+* CMake (version 3.4.2 and above)
+* OpenCL 1.2-capable hardware and drivers with SPIR 1.2 or SPIR-V support
+* OpenCL ICD Loader
+* OpenCL headers
+* gcc (version 5.4 and above)
+* [ComputeCpp](https://developer.codeplay.com)
+
+### Building SYCL-BLAS
+
+First get OpenBLAS. Instructions on building and installing this are on 
+the [project website](https://www.openblas.net/).
+
+```bash
+git clone https://github.com/xianyi/OpenBLAS.git
+make
+make install
+```
+
+SYCL-BLAS uses CMake as its build system. All the configuration options 
+can be found in the main CMakeLists.txt for the project and will show 
+up in CMake GUI if you use it.
+
+You will need to provide the location of the ComputeCpp install you are
+using with the variable `ComputeCpp_DIR`. It should point to the folder
+where `bin/`, `lib/` etc. are. This should be the only argument that is
+mandatory, everything else should be optional. The default build type is
+Release, though this can be overridden.
+
+The following command shows how to compile SYCL-BLAS. If you have installed OpenBLAS 
+in a different location adapt the path
+
+* set `ComputeCpp_DIR` to the ComputeCpp root path
+
+You may also need to tell CMake where to find the OpenBLAS binaries using the 
+```-DCMAKE_PREFIX_PATH=/opt/OpenBLAS``` argument
+
+```bash
+git clone --recursive  https://github.com/codeplaysoftware/sycl-blas.git
+mkdir build && cd build
+cmake ../ -DComputeCpp_DIR=/path/to/computecpp
+make
+```
+
+To install the SYCL-BLAS library
+
+```bash
+make install
+```
+
+Doxygen documentation can be generated by running:
+
+```bash
+doxygen doc/Doxyfile
+```
+
+### Sample Code
+
+The "samples" directory contains some sample code using SYCL-BLAS 
+ to implement GEMM and GEMV operations.
+
+#### How to compile and run the samples
+
+The "samples" folder contains a basic CMake configuration file and a module to find
+SYCL-BLAS (which will be used as a header-only framework).
+
+* set `ComputeCpp_DIR` to the ComputeCpp root path
+* set `SyclBLAS_DIR` to the SYCL-BLAS project root path
+
+```bash
+mkdir build
+cd build
+cmake .. -DComputeCpp_DIR=/path/to/computecpp \
+                 -DSyclBLAS_DIR=~/path/to/syclblas
+make
+```
+
+Execute the samples using the following commands
+```bash
+./gemm
+```
+
+or
+
+```bash
+./gemv
+```
+
+## Support
+
+### Bug reports and Issues
```

Comment:
Maybe this is beyond this PR, but I think this sort of information belongs more in CONTRIBUTING than README.

Reply:
My plan is to have this section for all our GitHub projects in the main README so that people understand what information to provide. Alternatively we can implement a template for issues that are raised.